### PR TITLE
Fix: Don't redirect twice on activation or update 

### PR DESCRIPTION
**Pull request by dereksmart**
Might just fix all of these issues: 
#1868, #1846, #1840

But will definitely fix: #1846

- Adds a param to `jetpack_pre_activate_module` that allows us to control a redirect after module activation.

In 3.4, we started calling a new method `do_version_bump()` <a href="https://github.com/Automattic/jetpack/blob/master/jetpack.php#L63">#</a> on every update or activation.  
This then triggered `activate_manage`, 
then `activate_module` (916064c), 
then the `jetpack_pre_activate_module` hook, 
then the `fix_redirect()` method <a href="https://github.com/Automattic/jetpack/blob/f9d70c1d0aa9a1f6befee5aa1506bb5b706acd81/class.jetpack-admin.php#L200">#</a> , 
which had already been called, resulting in multiple redirects. 

Since we're already sending over a false $redirect param from `activate_manage()` <a href="https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L391">#</a> , this PR will stop that method from triggering the redirect in `fix_redirect()`.  

TL;DR - we were redirecting twice on plugin activation and updates, and this puts a stop to it.

### CODE REVIEW

**Comment by samhotchkiss** [comment]
giphy

**Comment by zinigor** [comment]
Read the changes, tested with a fresh Jetpack installation - I can no longer reproduce #1846. Looks good to go!
